### PR TITLE
Use slices instead of Vectors

### PR DESCRIPTION
**Pull request by nyovaya**

### CODE REVIEW

**Comment by nyovaya** [comment]
How do I format it with the projects coding style? cargo fmt seems to use my own preset.

File: examples/testlibpq3.rs [hunk 1]
```diff
@@ -61,7 +61,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
      */
 
     /* Here is our out-of-line parameter value */
-    let param_values = vec![Some(b"joe's place\0".to_vec())];
+    let param_values = vec![Some("joe's place\0".as_bytes())];
```
Comment (sanpii) [inline examples/testlibpq3.rs:64]: Cna you use the same form as other occurences: `b"..."`?

**Comment by sanpii** [comment]
> How do I format it with the projects coding style? cargo fmt seems to use my own preset.

I use the default configurarion, normally nothing special. Maybe you have a global configuration (in a parent directory) ?

**Comment by nyovaya** [comment]
I doubt the valgrind error is caused by my changes :thinking:

**Comment by sanpii** [comment]
> I doubt the valgrind error is caused by my changes 🤔

You probably right, on my PC, the result is almost the same.

**Comment by nyovaya** [comment]
Is something missing here? @sanpii

**Comment by sanpii** [comment]
I fixed the CI, can you base your PR on my `main` branch?

I also tried to understand the complicated syntax for `param_values` in `examples/testlibpq3.rs`.

Maybe this form is clearer:
```rust
let param_values = [Some(b"joe's place\0".as_ref())];
```

In any cases, a vector is useless here.

**Comment by sanpii** [comment]
Please rebase, don't merge.

**Comment by nyovaya** [comment]
Done.

**Comment by sanpii** [comment]
Merged, thank you.

**Pull request closed without merging**